src/test/java/com/yegor256/tojos/MnCsvTest.java

```diff
@@ -122,4 +128,65 @@ void putsKeyFirst(@Mktmp final Path temp) throws IOException {
             )
         );
     }
+
+    @Test
+    void handlesConcurrentModificationInDupMethod(@Mktmp final Path temp) {
+        final Collection<Map<String, String>> rows = new ArrayList<>(10_000);
+        final AtomicReference<Throwable> error = new AtomicReference<>();
+        final Path path = temp.resolve("key-test.json");
+        final Mono csv = new MnCsv(path);
+        final int threads = 2;
+        MatcherAssert.assertThat(
+            "works without concurrency conflicts",
+            new Together<>(
+                threads, thread -> {
+                switch (thread) {
+                    case 0:
+                        add(rows);
+                        break;
+                    case 1:
+                        dup(csv, rows, error);
+                        break;
+                    default:
+                        break;
+                }
+                return thread;
+            }).asList(), Matchers.hasSize(2)
+        );
+        if (error.get() != null) {
+            final Throwable err = error.get();
+            Assertions.fail(
+                String.format(
+                    "Test failed due to exception: %s",
+                    err.getClass().getSimpleName()
+                ),
+                err
+            );
+        }
+    }
+
+    private static void add(final Collection<Map<String, String>> rows) {
+        for (int idx = 0; idx < 10_000; idx += 1) {
+            rows.add(
+                new HashMap<>(
+                    Collections.singletonMap(
+                        String.format("key%d", idx),
+                        String.format("value%d", idx)
+                    )
+                )
+            );
+        }
+    }
+
+    private static void dup(final Mono csv,
+        final Collection<Map<String, String>> rows,
+        final AtomicReference<Throwable> exc) throws InterruptedException {
+        Thread.sleep(50);
+        try {
+            csv.write(rows);
+        } catch (final ConcurrentModificationException ex) {
+            exc.set(ex);
```

Comment:
@VladMilash Do we need to collect all exceptions in a separate collection? If we get exception in this code:
```java
csv.write(rows)
```
a thread from the [together](https://github.com/yegor256/together) library will fail and the entire test should fail as well:

> Here, the Together class will run the "job" in multiple threads and will make sure that all of them return true. If at least one of them returns false, the test will fail. If at least one of the threads will throw an exception, the test will also fail.

(seems, we don't need `try-catch` here)